File: classical_algorithms/c++/bubblesort.cpp
#include<bits/stdc++.h>
using namespace std;

void BubSort(int a[],int n){
    cout<<"Bubble Sort: ";
    for(int i=0;i<n-1;i++){     //total N-1 paas
        int temp;
        for(int j=0;j<n-i-1;j++){   //elements to be traversed are reduced on every pass
            if(a[j]>a[j+1]){
                temp=a[j];
                a[j]=a[j+1];
                a[j+1]=temp;}
        }
    }
}

int main(){
    int arr[]={10,-4,7,3,1,0,2,6,99};
    int n=sizeof(arr)/sizeof(arr[0]);
    BubSort(arr,n);
    
    for(int i=0;i<n;i++)
        cout<<arr[i]<<" ";
}

File: classical_algorithms/dart/BubbleSort.dart
void main() {
  List<int> array = <int>[10, -4, 7, 3, 1, 0, 2, 6, 99];
  BubbleSort(array);

  for (int i = 0; i < array.length - 1; i++) print(array[i]);
}

List<int> BubbleSort(List<int> list) {
  print("Bubble Sort: ");
  for (int i = 0; i < list.length; i++) {
    for (int j = 0; j < list.length - 1; j++) {
      if (list[j] > list[j + 1]) {
        int num = list[j];
        list[j] = list[j + 1];
        list[j + 1] = num;
      }
    }
  }
  return list;
}
